src/java/org/apache/cassandra/index/sai/disk/PostingListRangeIterator.java

```diff
@@ -90,7 +90,7 @@ public PostingListRangeIterator(IndexContext indexContext,
     @Override
     protected void performSkipTo(PrimaryKey nextKey)
     {
-        if (skipToToken != null && skipToToken.compareTo(nextKey) >= 0)
```

Comment:
Add a comment explaining why?

Reply:
I didn't add a comment because @mike-tr-adamson mentioned this is the way it was fixed upstream. However, I cannot find that fix anywhere. I'll add a comment.

Reply:
https://github.com/apache/cassandra/pull/2943 is where this change is being made in OSS

Reply:
You might also rename skipToToken -> skipToKey since it is not actually a token currently.